macros/src/main/scala/macros.scala

```diff
@@ -185,7 +185,7 @@ private object MacroImpl {
       val constructorParams = constructor.paramss.head
 
       val tuple = Ident(newTermName("tuple"))
-      val (optional, required) = constructorParams.filterNot(ignoreField).zipWithIndex zip types partition (t => isOptionalType(t._2))
```

Comment:
Or *maybe* there is something wrong elsewhere with this PR. I confirm that running your tests without the change doesn't raise failure. Until the test is repeatable, this won't be merged.